standalone/test/integration_tests/api/integrationtest_scanner_api.cpp

```diff
@@ -90,7 +90,8 @@ class UserCallbacks
     util::Barrier start_req_barrier;                                                                                   \
     std::future<void> start_future;                                                                                    \
     EXPECT_START_REQUEST_CALL(*hw_mock, *config).WillOnce(OpenBarrier(&start_req_barrier));                            \
-    EXPECT_DOES_NOT_BLOCK(start_future = driver->start(););                                                            \
+    EXPECT_DOES_NOT_BLOCK_NOR_THROW(start_future = driver->start(););                                                  \
+    EXPECT_TRUE(start_future.valid());                                                                                 \
```

Comment:
Which cases prevent us from putting this expectation into the EXPECT DOES NOT BLOCK define?

Reply:
The EXPECT_DOES_NOT_BLOCK does not care about the internals of a statement.

Reply:
ah thats a different future, mixed that up